### ISSUE TITLE
Universal link outside status won't trigger status app open

### ISSUE THREAD

**Issue by yqrashawn** [status-im/status-react#18490]
# Problem

Universal link **outside** status won't trigger status app open
This issue here is to what works, since things break randomly when there's no change in iOS entitlements file and Android Manifest.xml file
Please go to the notion file, change the checkbox or add a comment if things don't work.

Note and Scope: 
- universal links are links started with `https://status.app`
- `status-app://` links are deep links, which works well
- Link in side app is out of scope
- If links are handled correctly after the app is opened, is out of scope
- App is installed
- Only one app is installed (Debug or PR or Release build)
- Helper page for testing https://yqrashawn.com/_test/sul.html
- Try Figma's universal link, and its mobile app see if anything should work
	- example Figma universal link https://www.figma.com/file/q3kZPl7OpbB7btFK7HG0Mr/Sharing-for-Mobile?type=design&node-id=2373-349410&mode=design&t=beaFKoN7QbK1ocPv-4

# Current behavior
[Tracked in notion since it's easier to comment and track on each line there](https://www.notion.so/Universal-link-outside-app-won-t-trigger-app-open-feef7657dcc64017ba9c75e176533597?pvs=4)

# Ever worked after migrated to the new link (join.status.im -> status.app)

- iOS
	- [ ] Link pasted into **Safari address bar** should trigger app open
		- [x] Release build
          - [x] Simulator
          - [x] Real device
		- [x] PR build
          - [x] Simulator
          - [x] Real device
		- [ ] Debug build
          - [ ] Simulator
          - [ ] Real device
	- [ ] Link pressed on a **webpage inside Safari** should trigger app open
		- [x] Release build
          - [x] Simulator
          - [x] Real device
		- [x] PR build
          - [x] Simulator
          - [x] Real device
		- [ ] Debug build
          - [ ] Simulator
          - [ ] Real device
	- [ ] Link pressed in **Telegram message view** should trigger app open
		- [ ] Release build
          - [ ] Simulator
          - [ ] Real device
		- [ ] PR build
          - [ ] Simulator
          - [ ] Real device
		- [ ] Debug build
          - [ ] Simulator
          - [ ] Real device
	- [ ] Link pressed in **Telegram in-app browser** should trigger app open
		- [ ] Release build
          - [ ] Simulator
          - [ ] Real device
		- [ ] PR build
          - [ ] Simulator
          - [ ] Real device
		- [ ] Debug build		
          - [ ] Simulator
          - [ ] Real device
- Android
	- [ ] Link pasted into **Chrome address bar** should trigger app open
		- [ ] Release build
          - [ ] Simulator
          - [x] Real device
		- [ ] PR build
          - [ ] Simulator
          - [x] Real device
		- [ ] Debug build
          - [ ] Simulator
          - [ ] Real device
	- [ ] Link pressed on a **webpage inside Chrome** should trigger app open
		- [ ] Release build
          - [ ] Simulator
          - [x] Real device
		- [ ] PR build
          - [ ] Simulator
          - [x] Real device
		- [ ] Debug build
          - [ ] Simulator
          - [ ] Real device
	- [ ] Link pressed in **Telegram message view** should trigger app open
		- [ ] Release build
          - [ ] Simulator
          - [ ] Real device
		- [ ] PR build
          - [ ] Simulator
          - [ ] Real device
		- [ ] Debug build
          - [ ] Simulator
          - [ ] Real device
	- [ ] Link pressed in **Telegram in-app browser** should trigger app open
		- [ ] Release build
          - [ ] Simulator
          - [ ] Real device
		- [ ] PR build
          - [ ] Simulator
          - [ ] Real device
		- [ ] Debug build				
          - [ ] Simulator
          - [ ] Real device

# Related Issues, PRs, comments
- https://github.com/status-im/status-mobile/issues/17550
- https://github.com/status-im/status-mobile/issues/18177
- https://github.com/status-im/status-mobile/pull/18363#issuecomment-1887156360
- https://github.com/status-im/status-mobile/pull/17480
- https://github.com/status-im/status-mobile/pull/18020#issuecomment-1850052276

# Resources

## App bundle identifier
- iOS
	- Release `im.status.ethereum`
	- PR `im.status.ethereum.pr`
	- Debug `im.status.ethereum.debug`
- Android
	- Release `im.status.ethereum`
	- PR `im.status.ethereum.pr`
	- Debug `im.status.ethereum`

## App association link
- iOS 
	- https://status.app/.well-known/apple-app-site-association
	- Third party validator https://yurl.chayev.com/ios-results?url=https://status.app/u/G10A4B0JdgwyRww90WXtnP1oNH1ZLQNM0yX0Ja9YyAMjrqSZIYINOHCbFhrnKRAcPGStPxCMJDSZlGCKzmZrJcimHY8BbcXlORrElv_BbQEegnMDPx1g9C5VVNl0fE4y&prefix=&bundle=
- Android 
	- https://status.app/.well-known/assetlinks.json
	- Third party validator https://yurl.chayev.com/android-results?url=https://status.app/u/G10A4B0JdgwyRww90WXtnP1oNH1ZLQNM0yX0Ja9YyAMjrqSZIYINOHCbFhrnKRAcPGStPxCMJDSZlGCKzmZrJcimHY8BbcXlORrElv_BbQEegnMDPx1g9C5VVNl0fE4y&prefix=&bundle=
## Universal link config
- iOS
	- Release and Debug https://github.com/status-im/status-mobile/blob/75ab9fdcc513b79881644cfceba6e68304739b35/ios/StatusIm/StatusIm.entitlements#L14
	- PR https://github.com/status-im/status-mobile/blob/75ab9fdcc513b79881644cfceba6e68304739b35/ios/StatusImPR/StatusImPR.entitlements#L14
- Android https://github.com/status-im/status-mobile/blob/75ab9fdcc513b79881644cfceba6e68304739b35/android/app/src/main/AndroidManifest.xml#L65

## How to get universal links

https://github.com/status-im/status-mobile/assets/15090582/08dfb33e-6e58-43b7-b93d-d1eabf233943

**Comment by felicio** [comment]
@yqrashawn just wanted to ask, based on https://github.com/status-im/status-mobile/issues/15325#issuecomment-2121836746, was this issue done as well?

**Comment by yqrashawn** [comment]
> @yqrashawn just wanted to ask, based on [#15325 (comment)](https://github.com/status-im/status-mobile/issues/15325#issuecomment-2121836746), was this issue done as well?

No, currently the universal link works on release and PR build, but won't work on debug build
